python/cudf/cudf/core/buffer/utils.py

```diff
@@ -71,6 +73,12 @@ def as_buffer(
             "`data` is a buffer-like or array-like object"
         )
 
+    if get_option("copy_on_write"):
+        if isinstance(data, (Buffer, RefCountableBuffer)) or hasattr(
```

Comment:
Every `RefCountableBuffer` is a `Buffer` by inheritance:

```suggestion
        if isinstance(data, Buffer) or hasattr(
```

(code change suggestion may fail style check)